### PR TITLE
Use the shorthand notation when applicable

### PR DESCRIPTION
**Pull request by greg0ire**
It's part of the standard

### CODE REVIEW

File: components/filesystem/lock_handler.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Usage
 
 A lock can be used, for example, to allow only one instance of a command to run.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/filesystem/lock_handler.rst]: Can we move this to the end of the previous line? Thanks!

File: components/expression_language/extending.rst [hunk 2]
```diff
@@ -70,7 +70,7 @@ which returns an array of expression functions (instances of
 :class:`Symfony\\Component\\ExpressionLanguage\\ExpressionFunction`) to
 register.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/expression_language/extending.rst]: Let's move this to the end of the previous line.

File: components/process.rst [hunk 3]
```diff
@@ -312,7 +312,7 @@ Process Pid
 You can access the `pid`_ of a running process with the
 :method:`Symfony\\Component\\Process\\Process::getPid` method.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/process.rst]: Merge with the previous line, please.

File: components/templating.rst [hunk 4]
```diff
@@ -190,7 +190,7 @@ choose which one to use for the template, the
 :method:`EngineInterface::supports() <Symfony\\Component\\Templating\\EngineInterface::supports>`
 method is used.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/templating.rst]: Merge with previous line, please.

File: components/translation.rst [hunk 5]
```diff
@@ -31,7 +31,7 @@ Configuration
 
 The constructor of the ``Translator`` class needs one argument: The locale.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/translation.rst]: Merge with previous line, please.

File: components/yaml.rst [hunk 6]
```diff
@@ -98,7 +98,7 @@ Reading YAML Files
 The :method:`Symfony\\Component\\Yaml\\Yaml::parse` method parses a YAML
 string and converts it to a PHP array:
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/yaml.rst]: Merge with previous line, please.

File: components/yaml.rst [hunk 7]
```diff
@@ -181,7 +177,7 @@ The second argument of the :method:`Symfony\\Component\\Yaml\\Yaml::dump`
 method customizes the level at which the output switches from the expanded
 representation to the inline one:
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline components/yaml.rst]: Merge with previous line, please.

File: configuration/external_parameters.rst [hunk 8]
```diff
@@ -186,7 +186,7 @@ to set. This is useful when important configuration is in a non-standard
 format. The example below includes a Drupal database configuration in
 the Symfony service container.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline configuration/external_parameters.rst]: Merge with previous line, please.

File: controller.rst [hunk 9]
```diff
@@ -310,7 +310,7 @@ object, which ultimately triggers a 404 HTTP response inside Symfony.
 Of course, you're free to throw any ``Exception`` class in your controller -
 Symfony will automatically return a 500 HTTP response code.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline controller.rst]: Merge with previous line, please.

File: controller/soap_web_service.rst [hunk 10]
```diff
@@ -94,7 +94,7 @@ Below is an example of a controller that is capable of handling a SOAP
 request. If ``indexAction()`` is accessible via the route ``/soap``, then the
 WSDL document can be retrieved via ``/soap?wsdl``.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline controller/soap_web_service.rst]: Merge with previous line, please.

File: doctrine/repository.rst [hunk 11]
```diff
@@ -60,7 +60,7 @@ Next, add a new method - ``findAllOrderedByName()`` - to the newly-generated
 ``ProductRepository`` class. This method will query for all the ``Product``
 entities, ordered alphabetically by name.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline doctrine/repository.rst]: Merge with previous line, please.

File: email/dev_environment.rst [hunk 12]
```diff
@@ -97,7 +97,7 @@ via the ``delivery_addresses`` option:
 
 Now, suppose you're sending an email to ``recipient@example.com``.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline email/dev_environment.rst]: Merge with previous line, please.

File: form/validation_group_service_resolver.rst [hunk 13]
```diff
@@ -41,7 +41,7 @@ parameter.
 
 Then in your form, inject the resolver and set it as the ``validation_groups``.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline form/validation_group_service_resolver.rst]: Merge with previous line, please.

File: form/validation_group_service_resolver.rst [hunk 14]
```diff
@@ -6,7 +6,7 @@ can't be determined by a simple callback, you can use a service. Create a
 service that implements ``__invoke()`` which accepts a ``FormInterface`` as a
 parameter.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline form/validation_group_service_resolver.rst]: Merge with previous line, please.

File: http_cache/esi.rst [hunk 15]
```diff
@@ -97,7 +97,7 @@ Now, suppose you have a page that is relatively static, except for a news
 ticker at the bottom of the content. With ESI, you can cache the news ticker
 independent of the rest of the page.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline http_cache/esi.rst]: Merge with previous line, please.

File: http_cache/esi.rst [hunk 16]
```diff
@@ -193,7 +193,7 @@ used ``render()``.
 The embedded action can now specify its own caching rules, entirely independent
 of the master page.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline http_cache/esi.rst]: Merge with previous line, please.

File: logging/processors.rst [hunk 17]
```diff
@@ -16,7 +16,7 @@ Sometimes it is hard to tell which entries in the log belong to which session
 and/or request. The following example will add a unique token for each request
 using a processor.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline logging/processors.rst]: Merge with previous line, please.

File: reference/constraints/Choice.rst [hunk 18]
```diff
@@ -131,7 +131,7 @@ if you want to keep your choices in some central location so that, for example,
 you can easily access those choices for validation or for building a select
 form element.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline reference/constraints/Choice.rst]: Merge with previous line, please.

File: reference/constraints/Valid.rst [hunk 19]
```diff
@@ -24,7 +24,7 @@ In the following example, create two classes ``Author`` and ``Address``
 that both have constraints on their properties. Furthermore, ``Author``
 stores an ``Address`` instance in the ``$address`` property.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline reference/constraints/Valid.rst]: Merge with previous line, please.

File: reference/forms/types/options/choice_value.rst.inc [hunk 20]
```diff
@@ -18,7 +18,7 @@ If you pass a callable, it will receive one argument: the choice itself. When us
 the :doc:`/reference/forms/types/entity`, the argument will be the entity object
 for each choice or ``null`` in some cases, which you need to handle:
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline reference/forms/types/options/choice_value.rst.inc]: Merge with previous line, please.

File: reference/forms/types/options/data_class.rst.inc [hunk 21]
```diff
@@ -6,7 +6,7 @@ data_class
 This option is used to set the appropriate data mapper to be used by the
 form, so you can use it for any form field type which requires an object.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline reference/forms/types/options/data_class.rst.inc]: Merge with previous line, please.

File: routing/hostname_pattern.rst [hunk 22]
```diff
@@ -410,7 +410,7 @@ Testing your Controllers
 You need to set the Host HTTP header on your request objects if you want to get
 past url matching in your functional tests.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline routing/hostname_pattern.rst]: Merge with previous line, please.

File: security/access_denied_handler.rst [hunk 23]
```diff
@@ -45,7 +45,7 @@ This interface defines one method called ``handle()`` that implements the logic
 execute when access is denied to the current user (send a mail, log a message, or
 generally return a custom response).
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/access_denied_handler.rst]: Merge with previous line, please.

File: security/api_key_authentication.rst [hunk 24]
```diff
@@ -274,7 +274,7 @@ need to implement the :class:`Symfony\\Component\\Security\\Http\\Authentication
 Authenticator. This will provide a method ``onAuthenticationFailure()`` which
 you can use to create an error ``Response``.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/api_key_authentication.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 25]
```diff
@@ -60,7 +60,7 @@ this data across the security context. First, you'll create your token class.
 This will allow the passing of all relevant information to your authentication
 provider.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 26]
```diff
@@ -106,7 +106,7 @@ A security listener should handle the
 :class:`Symfony\\Component\\HttpKernel\\Event\\GetResponseEvent` event, and
 set an authenticated token in the token storage if successful.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 27]
```diff
@@ -202,7 +202,7 @@ Namely, the provider will verify the ``Created`` header value is valid within
 five minutes, the ``Nonce`` header value is unique within five minutes, and
 the ``PasswordDigest`` header value matches with the user's password.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 28]
```diff
@@ -311,7 +311,7 @@ provider and any configuration options available for it. First, you must
 create a class which implements
 :class:`Symfony\\Bundle\\SecurityBundle\\DependencyInjection\\Security\\Factory\\SecurityFactoryInterface`.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 29]
```diff
@@ -471,7 +471,7 @@ to service ids that do not exist yet: ``wsse.security.authentication.provider``
 Now that your services are defined, tell your security context about your
 factory in your bundle class:
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 30]
```diff
@@ -565,7 +565,7 @@ can have different timeout lengths.
 You will first need to edit ``WsseFactory`` and define the new option in
 the ``addConfiguration()`` method.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/custom_authentication_provider.rst [hunk 31]
```diff
@@ -585,7 +585,7 @@ contain a ``lifetime`` key, set to 5 minutes (300 seconds) unless otherwise
 set in the configuration. Pass this argument to your authentication provider
 in order to put it to use.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/custom_authentication_provider.rst]: Merge with previous line, please.

File: security/impersonating_user.rst [hunk 32]
```diff
@@ -236,7 +236,7 @@ the sticky locale:
 
     The listener implementation assumes your ``User`` entity has a ``getLocale()`` method.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/impersonating_user.rst]: Merge with previous line, please.

File: security/remember_me.rst [hunk 33]
```diff
@@ -238,7 +238,7 @@ in the controller could be secured using the service context.
 In the following example, the action is only allowed if the user has the
 ``IS_AUTHENTICATED_FULLY`` role.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/remember_me.rst]: Merge with previous line, please.

File: security/voters.rst [hunk 34]
```diff
@@ -38,7 +38,7 @@ A custom voter needs to implement
 or extend :class:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\AbstractVoter`,
 which makes creating a voter even easier.
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline security/voters.rst]: Merge with previous line, please.

File: session/locale_sticky_session.rst [hunk 35]
```diff
@@ -129,7 +129,7 @@ locale value before they are redirected to their first page.
 To do this, you need an event listener for the ``security.interactive_login``
 event:
 
-.. code-block:: php
+::
```
Comment (javiereguiluz) [inline session/locale_sticky_session.rst]: Merge with previous line, please.

**Review by javiereguiluz — APPROVED** [review]
@greg0ire thanks for taking care of this. I'm sorry for the amount of review comments (35!) but I wanted to mark all the places where we can get rid of `::`.

By the way, I'm not sure I like replacing `.. code-block:: php` by `::` when there's no previous line of text. The problem with `::` is that it looks strange and it's not easy to understand, whereas `.. code-block:: php` is beautiful and self-explanatory. Ping @xabbuh what do you think?

File: components/translation.rst (hunk 5 above)
Comment (greg0ire) [reply components/translation.rst]: This one will look a bit weird I think

File: components/filesystem/lock_handler.rst (hunk 1 above)
Comment (xabbuh) [reply components/filesystem/lock_handler.rst]: While reading this comment there is at least one important difference to point out that we did not discuss on Slack: The colon of the short-hand notation will be rendered in the final document. So we need to check too if that really makes sense with each example.

File: components/translation.rst [hunk 36]
```diff
@@ -29,9 +29,7 @@ catalogs*).
 Configuration
 ~~~~~~~~~~~~~
 
-The constructor of the ``Translator`` class needs one argument: The locale.
-
-.. code-block:: php
+The constructor of the ``Translator`` class needs one argument: The locale::
```
Comment (greg0ire) [inline components/translation.rst]: @javiereguiluz @xabbuh that one will look a bit weird, won't it?

Comment (xabbuh) [reply components/translation.rst]: I would stick with the explicit code block here

Comment (greg0ire) [reply components/translation.rst]: > The problem with :: is that it looks strange and it's not easy to understand, whereas .. code-block:: php is beautiful and self-explanatory. Ping @xabbuh what do you think?

> I would stick with the explicit code block here

I'll extrapolate and take this as an agreement with @javiereguiluz 's feelings about `::` at the beginning of a line.

Comment (greg0ire) [reply components/translation.rst]: But that probably means I should amend the standard.

**Comment by javiereguiluz** [comment]
@greg0ire please, tell me when you consider this ready for merge. Thanks!

**Comment by greg0ire** [comment]
I believe it's ready,but warning, I amended the standard

File: contributing/documentation/standards.rst [hunk 37]
```diff
@@ -13,8 +13,8 @@ Sphinx
 * Each line should break approximately after the first word that crosses the
   72nd character (so most lines end up being 72-78 characters);
 * The ``::`` shorthand is *preferred* over ``.. code-block:: php`` to begin a PHP
-  code block (read `the Sphinx documentation`_ to see when you should use the
-  shorthand);
+  code block unless it results in the marker being on its own line (read
+  `the Sphinx documentation`_ to see when you should use the shorthand);
```
Comment (greg0ire) [inline contributing/documentation/standards.rst]: @javiereguiluz here is the part I amended, please tell me what you think

Comment (javiereguiluz) [reply contributing/documentation/standards.rst]: Looks good to me!

**Comment by javiereguiluz** [comment]
Let's merge this PR too ... and let's wish there are not too many conflicts. Thanks for working on this @greg0ire.

**Comment by javiereguiluz** [comment]
This has been merged up to `master` branch. As said in other PR, if you want to do this for the other branches, you should do it in 2.8, wait for merge, then 3.4 and wait for merge, then 4.0 and wait for merge and finally, master and wait for merger. Thanks!

File: configuration/external_parameters.rst [hunk 38]
```diff
@@ -124,7 +124,7 @@ You can now reference these parameters wherever you need them.
 
         </container>
 
-    .. code-block:: php
+    ::
```
Comment (fabpot) [inline configuration/external_parameters.rst]: This change is wrong as this is part of a `configuration-block` block. Reverted in #9368.

Comment (greg0ire) [reply configuration/external_parameters.rst]: Woops, sorry for this!

**Pull request closed without merging**